src/tasks/barfTasks.ts

```diff
@@ -343,15 +344,55 @@ const turns: AdventureAction[] = [
       setChoice(536, choiceToSet);
       ensureEffect($effect`Transpondent`);
       use($item`Map to Safety Shelter Grimace Prime`);
-      return true;
     },
+    completed: () => true,
     spendsTurn: true,
     sobriety: Sobriety.DRUNK,
   },
+  {
+    name: "Thesis",
+    ready: needTurns,
+    do: deliverThesisIfAble,
+    completed: () => !needTurns(),
+    sobriety: Sobriety.SOBER,
+    spendsTurn: false,
+  },
+  {
+    name: "Sausages",
+    ready: () =>
+      needTurns() &&
+      have($item`Kramco Sausage-o-Matic™`) &&
+      (have($item`magical sausage`) || have($item`magical sausage casing`)) &&
+      get("_sausagesEaten") < 23,
+    do: (): void => {
+      const available = clamp(
+        23 - get("_sausagesEaten"),
+        0,
+        itemAmount($item`magical sausage`) + itemAmount($item`magical sausage casing`)
+      );
+      eat(available, $item`magical sausage`);
+    },
+    completed: () => !needTurns(),
```

Comment:
```suggestion
    completed: () =>get("_sausagesEaten") >= 23 || (!have($item`magical sausage casing`) && !have($item`magical sausage`)),
```
Also maybe we should throw in logic where we get dressed in +mp gear here, idk